api/v1alpha1/flowmetric_types.go

```diff
@@ -0,0 +1,141 @@
+	// +required
+	Type MetricType `json:"type"`
+
+	// `valueField` is the flow field that must be used as a value for this metric. This field must hold numeric values.
+	// Leave empty to count flows rather than a specific value per flow.
+	// Refer to the documentation for the list of available fields: https://docs.openshift.com/container-platform/latest/networking/network_observability/json-flows-format-reference.html.
+	// +optional
+	ValueField string `json:"valueField,omitempty"`
+
+	// `filters` is a list of fields and values used to restrict which flows are taken into account. Oftentimes, these filters must
+	// be used to eliminate duplicates: `Duplicate:"false"` and `FlowDirection: "0"`.
+	// Refer to the documentation for the list of available fields: https://docs.openshift.com/container-platform/latest/networking/network_observability/json-flows-format-reference.html.
+	// +optional
+	Filters []MetricFilter `json:"filters"`
+
+	// `labels` is a list of fields that should be used as Prometheus labels, also known as dimensions.
+	// From choosing labels results the level of granularity of this metric, as well as the available aggregations at query time.
+	// It must be done carefully as it impacts the metric cardinality (cf https://rhobs-handbook.netlify.app/products/openshiftmonitoring/telemetry.md/#what-is-the-cardinality-of-a-metric).
+	// In general, avoid setting very high cardinality labels such as IP or MAC addresses.
+	// "SrcK8S_OwnerName" or "DstK8S_OwnerName" should be preferred over "SrcK8S_Name" or "DstK8S_Name" as much as possible.
+	// Refer to the documentation for the list of available fields: https://docs.openshift.com/container-platform/latest/network_observability/json-flows-format-reference.html.
+	// +optional
+	Labels []string `json:"labels"`
+
+	// When set to `true`, flows duplicated across several interfaces will add up in the generated metrics.
+	// When set to `false` (default), it is equivalent to adding the exact filter on `Duplicate`: `false`.
+	// +optional
+	IncludeDuplicates bool `json:"includeDuplicates,omitempty"`
+
+	// Filter for ingress, egress or any direction flows.
+	// When set to `Ingress`, it is equivalent to adding the regex filter on `FlowDirection`: `0|2`.
+	// When set to `Egress`, it is equivalent to adding the regex filter on `FlowDirection`: `1|2`.
+	// +kubebuilder:validation:Enum:="Any";"Egress";"Ingress"
+	// +kubebuilder:default:="Any"
+	// +optional
+	Direction FlowDirection `json:"direction,omitempty"`
+
+	// A list of buckets to use when `type` is "Histogram". The list must be parseable as floats. Prometheus default buckets will be used if unset.
+	// +optional
+	Buckets []string `json:"buckets,omitempty"`
+}
+
+// FlowMetricStatus defines the observed state of FlowMetric
+type FlowMetricStatus struct {
+	// INSERT ADDITIONAL STATUS FIELD - define observed state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+}
```

Comment:
It would be great to have a validation result here in case something went wrong.
At first maybe just report if `flowMetricToFLP` returned an error ?

In the future we could improve to check `valueField`, `labels` and `filters` are correct for example.

Reply:
This looks like something to do in a validation webhook, no?
I am creating a story for it: https://issues.redhat.com/browse/NETOBSERV-1447